From 210431852f0f0d7b4d591b13517a4c877e18aadc Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 5 Jun 2015 16:48:10 +0200
Subject: [PATCH] base: add "hostname=<hostname>" kernel cmdline parameter

For e.g. lvm, the hostname has to be set, so let there be a kernel
command line parameter for it.
---
 dracut.8                           |  3 +++
 modules.d/99base/install           |  1 +
 modules.d/99base/parse-hostname.sh | 12 ++++++++++++
 3 files changed, 16 insertions(+)
 create mode 100755 modules.d/99base/parse-hostname.sh

diff --git a/dracut.8 b/dracut.8
index 3c58453..02d3947 100644
--- a/dracut.8
+++ b/dracut.8
@@ -146,6 +146,9 @@ do not honor special mount options for the root filesystem found in
 
 .SS Misc
 .TP
+.BR hostname= "<hostname>"
+set the initial hostname
+.TP
 .BR rdblacklist= "<drivername>"
 do not load kernel module <drivername>
 This parameter can be specified multiple times.
diff --git a/modules.d/99base/install b/modules.d/99base/install
index 4a4531a..4227a5a 100755
--- a/modules.d/99base/install
+++ b/modules.d/99base/install
@@ -24,6 +24,7 @@ else
 	|| derror "Failed to install switch_root"
 fi
 inst "$moddir/dracut-lib.sh" "/lib/dracut-lib.sh"
+inst_hook cmdline 01 "$moddir/parse-hostname.sh"
 inst_hook cmdline 10 "$moddir/parse-root-opts.sh"
 inst_hook cmdline 20 "$moddir/parse-blacklist.sh"
 if [ -x "/usr/sbin/load_policy" -o -x "/sbin/load_policy" ]; then
diff --git a/modules.d/99base/parse-hostname.sh b/modules.d/99base/parse-hostname.sh
new file mode 100755
index 0000000..009ac96
--- /dev/null
+++ b/modules.d/99base/parse-hostname.sh
@@ -0,0 +1,12 @@
+type hostname >/dev/null 2>&1 || \
+    hostname() {
+        if [ -n "$1" ]; then
+            printf -- "%s" "$1" > /proc/sys/kernel/hostname
+        else
+            cat /proc/sys/kernel/hostname
+        fi
+}
+
+if hname=$(getarg hostname=); then
+    hostname "$hname"
+fi
